src/main/java/uk/gov/companieshouse/limitedpartnershipsapi/controller/LimitedPartnerController.java

```diff
@@ -137,4 +140,30 @@ public ResponseEntity<Object> deleteLimitedPartner(@RequestAttribute(TRANSACTION
 
         return new ResponseEntity<>(HttpStatus.NO_CONTENT);
     }
+
+    @GetMapping("/limited-partner/{" + URL_PARAM_LIMITED_PARTNER_ID + "}/validation-status")
+    public ResponseEntity<ValidationStatusResponse> getValidationStatus(@RequestAttribute(TRANSACTION_KEY) Transaction transaction,
+                                                                        @PathVariable(URL_PARAM_LIMITED_PARTNER_ID) String limitedPartnerId,
+                                                                        @RequestHeader(value = ERIC_REQUEST_ID_KEY) String requestId)
+            throws ServiceException {
+
+        var logMap = new HashMap<String, Object>();
+        logMap.put(URL_PARAM_TRANSACTION_ID, transaction.getId());
+        logMap.put(URL_PARAM_LIMITED_PARTNER_ID, limitedPartnerId);
+
+        ApiLogger.infoContext(requestId, "Calling service to validate a Limited Partner", logMap);
+        var validationStatus = new ValidationStatusResponse();
+        validationStatus.setValid(true);
+
+        var validationErrors = limitedPartnerService.validateLimitedPartner(transaction, limitedPartnerId);
+
+        if (!validationErrors.isEmpty()) {
+            ApiLogger.errorContext(requestId, String.format("Validation errors: %s",
+                    new GsonBuilder().create().toJson(validationErrors)), null, logMap);
+            validationStatus.setValid(false);
+            validationStatus.setValidationStatusError(validationErrors.toArray(new ValidationStatusError[0]));
```

Comment:
validationErrors.toArray(ValidationStatusError[]::new) is a better way to convert apparently (Automatically resizes the array to match the list size)

Reply:
Done. Change also applied to other controllers where this was being done.